spark/src/test/scala/org/apache/spark/sql/CometTestBase.scala

```diff
@@ -116,12 +116,49 @@ abstract class CometTestBase
     require(absTol > 0 && absTol <= 1e-6, s"absTol $absTol is out of range (0, 1e-6]")
 
     actualAnswer.toSeq.zip(expectedAnswer.toSeq).foreach {
+      case (actual: Float, expected: Float) =>
+        def isPosInfinity(value: Float): Boolean = {
+          value.isPosInfinity || value == 3.4028235e38
+        }
+
+        if ((actual.isNaN && expected.isNaN) ||
+          (isPosInfinity(actual) && isPosInfinity(expected)) ||
+          (actual.isNegInfinity && expected.isNegInfinity)) {
+          // ok
+        } else {
+
+          def almostEqual(a: Double, b: Double, tolerance: Double = 1e-6f): Boolean = {
```

Comment:
Looks like there are multiple identical definitions of `almostEqual`?

Reply:
Yes, the code is hacky and experimental. Will clean up before marking as ready for review.

Reply:
Actually, the two implementations of `almostEqual` have different input types (float vs double).